python-client/wmill/wmill/client.py

```diff
@@ -1,527 +1,630 @@
-            sleep(2.0)
+        for id_ in job_ids:
+            result[id_] = self.post(
+                f"/w/{self.workspace}/jobs_u/queue/cancel/{id_}",
+                json={"reason": "killed by `cancel_running` method"},
+            )
+
+        return result
+
+    def get_job_status(self, job_id: str) -> JobStatus:
+        resp = self.get(f"/w/{self.workspace}/jobs_u/get/{job_id}", raise_for_status=False)
+        assert not resp.status_code == 404, f"{job_id} not found"
+        resp_json = resp.json()
+        job_type = resp_json.get("type", "")
+        assert job_type, f"{resp_json} is not a valid job"
+        if job_type.lower() == "completedjob":
+            return "COMPLETED"
+        additional_properties = resp_json.get("additional_properties", {})
+        if "running" not in additional_properties:
+            raise Exception(f"{job_id} is not running")
+        if additional_properties.get("running"):
+            return "RUNNING"
+        return "WAITING"
+
+    def get_result(
+        self,
+        job_id: str,
+        assert_result_is_not_none: bool = True,
+    ) -> Any:
+        result = self.get(f"/w/{self.workspace}/jobs_u/completed/get_result/{job_id}")
+        result_text = result.text
+        if assert_result_is_not_none and result_text is None:
+            raise Exception(f"result is None for {job_id = }")
+        try:
+            return result.json()
+        except JSONDecodeError:
+            return result_text
+
+    def get_variable(self, path: str) -> str:
+        """Get variable from Windmill"""
+        return self.get(f"/w/{self.workspace}/variables/get/{path}").json()["value"]
+
+    def set_variable(self, path: str, value: str) -> None:
+        """Set variable from Windmill"""
+        # check if variable exists
+        r = self.get(f"/w/{self.workspace}/variables/get/{path}", raise_for_status=False)
+        if r.status_code == 404:
+            # create variable
+            self.post(
+                f"/w/{self.workspace}/variables/create",
+                json={
+                    "path": path,
+                    "value": value,
+                    "is_secret": False,
+                    "description": "",
+                },
+            )
         else:
-            sleep(5.0)
-        nb_iter += 1
-    
-    result = get_result(
-        job_id,
-        assert_result_is_not_none=assert_result_is_not_none,
-    )
+            # update variable
+            self.post(
+                f"/w/{self.workspace}/variables/update/{path}",
+                json={"value": value},
+            )
 
-    # the job finished--we don't need to cancel it anymore
-    if cleanup:
-        atexit.unregister(cancel_job)
+    def get_resource(
+        self,
+        path: str,
+        none_if_undefined: bool = False,
+    ) -> str | None:
+        """Get resource from Windmill"""
+        try:
+            return self.get(f"/w/{self.workspace}/resources/get/{path}").json()["value"]
```

Comment:
this should be using get_value_interpolated instead: https://app.windmill.dev/openapi.html#/operations/getResourceValueInterpolated